### PR TITLE
Added an option to use explicit dummies in AMBER

### PR DESCRIPTION
**Pull request by msuruzhon**
# Is this pull request to fix a bug, or to introduce new functionality?

This PR adds an option to handle dummies explicitly in AMBER.

## If this is to fix a bug...

This pull request fixes issue #?

* I confirm that I have merged the latest version of `devel` into this branch before issuing this pull request (e.g. by running `git pull origin devel`): [y/n]
* I confirm that I have permission to release this code under the GPL3 license: [y/n]

## If this introduces new functionality...

Changes proposed in this pull request:

* I confirm that I have merged the latest version of `devel` into this branch before issuing this pull request (e.g. by running `git pull origin devel`): [y/n]
* I confirm that I have added a test for any new functionality in this pull request: [y/n]
* I confirm that I have added documentation (e.g. a new tutorial page or detailed guide) for any new functionality in this pull request: [y/n]
* I confirm that I have permission to release this code under the GPL3 license: [y/n]

## Suggested reviewers:
@lohedges, @chryswoods

## Any additional context of information?

(you can also attach files by dragging & dropping)

### CODE REVIEW

**Review by chryswoods — APPROVED** [review]

File: python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py [hunk 1]
```diff
@@ -1706,29 +1713,46 @@ def _toRegularMolecule(
...
 
+        if generate_intrascale:
+            # First we regenerate the connectivity based on the bonds.
+            conn = _SireMol.Connectivity(mol.info()).edit()
+            for bond in mol.property("bond").potentials():
+                conn.connect(bond.atom0(), bond.atom1())
+            mol.setProperty("connectivity", conn.commit())
+
+            # Now we have the correct connectivity, we can regenerate the exclusions.
+            gro_sys = _SireIO.GroTop(_System(mol)._sire_object).toSystem()
```
Comment (xiki-tempula) [inline python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py:1753]: How does Gromacs come into play here?

File: test/Sandpit/Exscientia/Align/test_merge.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    proc.start()
+    proc.wait()
+
+    res = proc.getPotentialEnergy().value()
+
+    return res
+
+
+@pytest.fixture
+def benzene():
```
Comment (xiki-tempula) [inline test/Sandpit/Exscientia/Align/test_merge.py]: Would it be better if these move to conftest?

**Review by xiki-tempula — APPROVED** [review]
LGTM a couple samll questions

File: python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py (hunk 1 above)
Comment (msuruzhon) [reply python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py:1753]: This is just a hack to regenerate the exclusion list, I think it has nothing to do with Gromacs per se. Lester uses it in a few places so I have just copied the approach.

File: test/Sandpit/Exscientia/Align/test_merge.py (hunk 2 above)
Comment (chryswoods) [reply test/Sandpit/Exscientia/Align/test_merge.py]: I agree - ideally all fixtures should be in conftest so they don't get accidentally duplicated.

Comment (lohedges) [reply test/Sandpit/Exscientia/Align/test_merge.py]: Or just scope it. I've done this throughout, but should go back through and put duplicates in a conftest.

**Pull request merged**